### PR TITLE
zypperpkg: ignore retcode 104 for search()

### PR DESCRIPTION
**Pull request by aplanas**
Fix bsc#1159670

### CODE REVIEW

File: salt/modules/zypperpkg.py [hunk 1]
```diff
@@ -405,7 +405,9 @@ def _get_available_versions(self):
         Get available versions of the package.
         :return:
         '''
-        solvables = self.zypper.nolock.xml.call('se', '-xv', self.name).getElementsByTagName('solvable')
+        solvables = self.zypper.nolock.xml.call(
+            "se", "-xv", self.name, success_retcodes=[104]
```
Comment (isbm) [inline salt/modules/zypperpkg.py:409]: You already calling XML, do you still need "x" option?

File: salt/modules/zypperpkg.py [hunk 2]
```diff
@@ -983,7 +985,11 @@ def _is_match(pkgname):
         return False
 
     root = kwargs.get('root') or None
-    for node in __zypper__(root=root).xml.call('se', '-s', *targets).getElementsByTagName('solvable'):
+    for node in (
+        __zypper__(root=root)
+        .xml.call("se", "-s", *targets, success_retcodes=[104])
```
Comment (isbm) [inline salt/modules/zypperpkg.py:990]: Would be awesome not to have a magic hardcodee but define it somewhere as a constant above.

File: salt/modules/zypperpkg.py (hunk 1 above)
Comment (aplanas) [reply salt/modules/zypperpkg.py:409]: Seems that is not needed. Removed!

File: salt/modules/zypperpkg.py (hunk 2 above)
Comment (aplanas) [reply salt/modules/zypperpkg.py:990]: Good idea, yes.

**Review by isbm — APPROVED** [review]

**Comment by isbm** [comment]
👍 👍

**Pull request merged**